packages/react/src/lib/factories.ts

```diff
@@ -8,18 +8,19 @@ import {
   ShorthandRenderCallback,
   ShorthandRenderFunction,
   ShorthandRenderer,
+  WithAsProp,
 } from '../types'
 import { mergeStyles } from './mergeThemes'
 
 type HTMLTag = 'iframe' | 'img' | 'input'
 type ShorthandProp = 'children' | 'src' | 'type'
 
-interface CreateShorthandOptions {
+interface CreateShorthandOptions<P> {
   /** Default props object */
-  defaultProps?: Props
+  defaultProps?: P
 
   /** Override props object or function (called with regular props) */
-  overrideProps?: Props & ((props: Props) => Props) | Props
+  overrideProps?: P & ((props: P) => P) | P
```

Comment:
```suggestion
  overrideProps?: Partial<Props<P>> | ((props: P) => Partial<Props<P>>)
```

https://github.com/stardust-ui/react/pull/1886/files#r322256487